src/ui/components/Button/index.js

```diff
@@ -9,31 +9,32 @@ import './Button.scss';
 
 export default class Button extends React.Component {
   static propTypes = {
-    appearance: PropTypes.string,
```

Comment:
> I agree. Asking each caller to know the class name puts an undue mental burden on them (they have to look it up in the css or try to find an example of a similar call).

@kumar303 Heh it's interesting to see the different viewpoints. 

FWIW I argued in the opposite direction. Having multiple props (as this did earlier) can create ambiguity about what they affect unless it's very clear through naming what changes style versus other attributes of the button. 

Simply passing through the className prop unambiguously informs the developer what affects the styling and it makes clear the relationship between the JSX markup and the CSS without having to inspect it running on a page or drill into the component code to know what's what.

Using props to control styles used adds a layer of indirection as opposed to following a long established web convention.

If the props route is seen as the way forward I think it would be important to establish some  conventions around naming and have decent docs so it's easy for anyone to know what changes will provide the necessary result easily.

For a real-world example here's how react-bootstrap use props. https://react-bootstrap.github.io/components.html#buttons-props. It's not too bad if you have the docs to hand and they have some naming conventions to attempt make it a bit more intuitive.

It's worth trying it because (and maybe this is somewhat related to familiarity) but react-bootstrap seemed to be more annoying to use than the non-react version that just used structured classnames.